lib/file-source/src/file_watcher.rs

```diff
@@ -19,7 +20,7 @@ use crate::metadata_ext::PortableMetadataExt;
 pub struct FileWatcher {
     pub path: PathBuf,
     findable: bool,
-    reader: io::BufReader<fs::File>,
+    reader: Box<dyn BufRead>,
```

Comment:
is there a reason you choose a trait object over a regular generic?

Reply:
Ah I see here you could prob just do `impl Trait` and trait object below or you could be even more fancy and use an `Either` type of trait but that is probably overkill :)

Reply:
Yeah, there are actually two different types so an existential wouldn't work. One step fancier would an `Either`-like enum, which I'll probably do if we decide to implement `Seek` somewhere down the line but didn't seem worthwhile right now.